File: app/src/main/java/gm/tieba/tabswitch/widget/NavigationBar.java
package gm.tieba.tabswitch.widget;

import android.view.View;
import android.widget.TextView;

import de.robv.android.xposed.XposedHelpers;
import gm.tieba.tabswitch.XposedContext;
import gm.tieba.tabswitch.util.ReflectUtils;

public class NavigationBar extends XposedContext {
    private final Object mNavigationBar;

    public NavigationBar(final Object thisObject) {
        mNavigationBar = ReflectUtils.getObjectField(thisObject,
                "com.baidu.tbadk.core.view.NavigationBar");
    }

    public void addTextButton(final String text, final View.OnClickListener l) {
        final Class<?> ControlAlign = XposedHelpers.findClass(
                "com.baidu.tbadk.core.view.NavigationBar$ControlAlign", sClassLoader);
        for (final Object HORIZONTAL_RIGHT : ControlAlign.getEnumConstants()) {
            if (HORIZONTAL_RIGHT.toString().equals("HORIZONTAL_RIGHT")) {
                final TextView textView = (TextView) XposedHelpers.callMethod(mNavigationBar,
                        "addTextButton", HORIZONTAL_RIGHT, text, l);
                textView.setTextColor(ReflectUtils.getColor("CAM_X0105"));
                break;
            }
        }
    }

    public void setTitleText(final String title) {
        XposedHelpers.callMethod(mNavigationBar, "setTitleText", title);
    }
}

File: app/src/main/java/gm/tieba/tabswitch/widget/NavigationBar.kt
package gm.tieba.tabswitch.widget

import android.view.View
import android.widget.TextView
import de.robv.android.xposed.XposedHelpers
import gm.tieba.tabswitch.XposedContext
import gm.tieba.tabswitch.util.getColor
import gm.tieba.tabswitch.util.getObjectField

class NavigationBar(thisObject: Any) : XposedContext() {

    private val mNavigationBar: Any? = getObjectField(
        thisObject,
        "com.baidu.tbadk.core.view.NavigationBar"
    )

    fun addTextButton(text: String?, l: View.OnClickListener?) {
        val controlAlignClass = findClass("com.baidu.tbadk.core.view.NavigationBar\$ControlAlign")
        val horizontalRight = controlAlignClass.enumConstants.find { it.toString() == "HORIZONTAL_RIGHT" }
            ?: throw IllegalStateException("HORIZONTAL_RIGHT enum constant not found")
        val textView = XposedHelpers.callMethod(
            mNavigationBar,
            "addTextButton", horizontalRight, text, l
        ) as TextView
        textView.setTextColor(getColor("CAM_X0105"))
    }

    fun setTitleText(title: String?) {
        title?.let {
            XposedHelpers.callMethod(mNavigationBar, "setTitleText", it)
        }
    }

    fun setCenterTextTitle(title: String?) {
        title?.let {
            XposedHelpers.callMethod(mNavigationBar, "setCenterTextTitle", it)
        }
    }
}
